clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -226,6 +233,13 @@ class AzureNetworkClient extends AzureBaseClient {
     }
   }
 
+  private void addSecurityGroupToSubnet(AzureCredentials creds, String resourceGroupName, String securityGroupName, Subnet subnet) {
+    def securityGroup = getNetworkSecurityGroup(creds, resourceGroupName, securityGroupName)
+    def resourceId = new ResourceId()
+    resourceId.setId(securityGroup.resourceId)
+    subnet.networkSecurityGroup = resourceId
```

Comment:
We need to call SubnetOperations().createOrUpdate() in order to make the change stick to the subnet...